client/src/main/java/edu/colostate/cs/cs414/chesshireCoders/jungleClient/app/Game/GameBoard.java

```diff
@@ -0,0 +1,275 @@
+			if(!(((pieceColumn == 0) && (column == 3)) || ((pieceColumn == 3) && (column == 0)) || ((pieceRow == 2) && (row == 6)) || ((pieceRow == 6) && (row == 2)))) {
+				//if it is not jumping the right river return false;
+				if(!(((pieceColumn == 3) && (column == 6)) || ((pieceColumn == 6) && (column == 3)) || ((pieceRow == 2) && (row == 6)) || ((pieceRow == 6) && (row == 2)))) {
+					return false;
+				}
+			}
+			
+			//If the tiger or leopard is jumping the river there can not be a rat in the way.
+			//We first need to check where the animal is starting
+			int startingColumn = -1;
+			int startingRow = -1; 
+			if(pieceColumn < column) {
+				startingColumn = pieceColumn;
+			}
+			if(pieceColumn > column) {
+				startingColumn = column;
+			}
+			if(pieceRow < row) {
+				startingRow = row;				
+			}
+			if(pieceRow > row) {
+				startingRow = row;				
+			}	
+			
+			//Implement through the squares the animal travels to check if the rat occupies the same space.
+			if(startingColumn < 0) {
+				while(startingColumn < startingColumn+3) {
+					if((gamePieces[0][1].getColumn() == startingColumn)	&& (gamePieces[0][1].getRow()==row)) {
+						return false;
+					}
+					if((gamePieces[1][1].getColumn() == startingColumn)	&& (gamePieces[1][1].getRow()==row)) {
+						return false;
+					}
+					startingColumn++;
+				}
+			} else {
+				while(startingRow < startingRow+4) {
+					if((gamePieces[0][1].getColumn() == column)	&& (gamePieces[0][1].getRow() == startingRow)) {
+						return false;
+					}
+					if((gamePieces[1][1].getColumn() == column)	&& (gamePieces[1][1].getRow( )== startingRow)) {
+						return false;
+					}
+					startingRow++;
+				}
+			}
+		}
+		
+		//At this point the piece is either moving 1 space or it's a tiger/leopard jumping a river
+		//check if it can land on the space. 
+		//if it's normal, trap, or den anything can be on it.
+		BoardSquare boardSquare = boardSquares[column][row];
+		BoardSquare pieceSquare = boardSquares[gamePiece.getColumn()][gamePiece.getRow()];
+		boolean enemyBanned = false;
+		if(boardSquare.getSquareType().equals(SquareType.River)) {
+			//If it's not a normal, trap, or den it must be a river. Only rats can go on river. So if it's not a rat return false.
+			if(!(gamePiece.getPieceType().equals(PieceType.Rat))) {
+				return false;
+			}
+			//if is a rat and the rat is not already in the river their can not be a piece there.
+			if(pieceSquare.getSquareType().equals(SquareType.River)) {
+				enemyBanned = true;
+			}
+		}	
+		
+		//Check if their is a creature on it with a higher power
+		//if any of the player's pieces are on the space return false;
+		for(int i = 0; i < 8; i++) {
+			if((gamePieces[color][i].getRow() == row) && (gamePieces[0][i].getRow() == column)) {
+				return false;
+			}
+		}
+		
+		//if the space the piece is moved to is a FRIENDLY trap, any enemy will be consumed so just return true;
+		//is it a trap?
+		if (boardSquare.getSquareType().equals(SquareType.Trap)){
```

Comment:
This could be handled by the previous check. We just need to set the power of a `GamePiece` to 1 when it enters a trap and `piece.setPowerDefault()` when it exits.

Reply:
I can either make a method to control the current power of a piece  requiring an update each move, or I can have one check in this one method. It seemed simpler.